### DISCUSSION TITLE
Gdu conflicts with coreutils

### DISCUSSION THREAD

**Discussion by dundee** [Homebrew/discussions#4557] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 8917acac9560f8e2846eedc4181470f0345f6db8
Last commit: 6 days ago
Core tap JSON: 05 Jun 11:52 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.41.0 => /opt/homebrew/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hello community,
I am the maintainer of [gdu](https://formulae.brew.sh/formula/gdu). The current formulae has a [conflict with coreutils](https://github.com/Homebrew/homebrew-core/blob/eeb7acb8d1ea043265cd9177c96db9428d2dafa4/Formula/gdu.rb#LL20C19-L20C28), which was [reported](https://github.com/dundee/gdu/issues/48) by the users of gdu.

The current installation instructions work well:

```
brew install coreutils
brew install --force gdu
brew link --overwrite gdu
```

but the upgrade fails then.

Is there some recommended approach how to solve such situations? 

My ideas:
* change the name of the binary to smth like `gdu-compat`
* introduce new formulare, e.g. `gdu-compat`

**Comment by carlocab** [comment, marked as answer]
Users can `brew unlink coreutils` for `gdu` upgrades or `brew unlink gdu` for `coreutils` updates. They can do `brew link --overwrite` after that to restore the symlinks.

It's admittedly not a great UX, but we don't really have a nice way to handle these at the moment. The simplest thing to do may be to rename the binary to something like `go-gdu`, or `gdu-go`.

**Reply by dundee** [reply]
I will rename the binary. Thanks!

**Reply by dundee** [reply]
https://github.com/Homebrew/homebrew-core/pull/134668
